sample_apps/rso_sample_apps/rust/src/service.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{config, handlers};
+use futures::executor;
+use log::debug;
+use warp::Filter;
+
+/// Runs the web service
+///
+/// # Arguments
+///
+/// `cfg`: The configuration to use for the service.
+///
+/// # Returns
+///
+/// None
+///
+/// # Panics
+///
+/// Panics if the host address is invalid.
+pub(crate) fn listen(cfg: &config::Configuration) {
+    debug!("parsing server {}", cfg.server.addr());
+    match cfg.server.addr().parse::<std::net::SocketAddr>() {
+        Ok(addr) => {
+            debug!("parsed server {}", addr);
+            let routes = warp::get()
+                .and(handlers::oauth::handle(cfg))
+                .or(handlers::data::handle(cfg))
+                .or(handlers::default::handle(cfg));
+            match cfg.clone().server.tls {
+                Some(tls) => {
+                    executor::block_on(
```

Comment:
This should be an async function which does not spawn a blocking operation for each request that's handled.